analytical_engine/core/fragment/dynamic_fragment.h

```diff
@@ -491,24 +502,27 @@ class DynamicFragment
     is_selfloops_.clear();
   }
 
-  void CopyFrom(std::shared_ptr<DynamicFragment> source,
+    void CopyFrom(std::shared_ptr<DynamicFragment> source,
                 const std::string& copy_type = "identical") {
     init(source->fid_, source->directed_);
     load_strategy_ = source->load_strategy_;
     copyVertices(source);
 
+    // both inner and outer vertices with the empty slots
+    auto vnum = id_parser_.max_local_id();
```

Comment:
This did not make sense since the vnum would be very very huge (the max_local_id could be std::unlimited<uint64>`

Reply:
Would it be better to split the loop into two (one for inner vertices, the other for outer vertices)?

Reply:
> This did not make sense since the `vnum` could be very very huge (the max_local_id could be std::unlimited<uint64>`

Would it be better to split the loop into two (one for inner vertices, the other for outer vertices)?

Reply:
Fixed by skipping the empty gap and splitting the degree vector